internal/backends/winit/renderer/sw.rs

```diff
@@ -90,53 +79,19 @@ impl<const MAX_BUFFER_AGE: usize> super::WinitCompatibleRenderer
             });
         }
 
-        let image_ref: imgref::ImgRef<rgb::RGB8> =
-            imgref::ImgRef::new(&buffer, width, height).into();
-
-        canvas.opengl_context.make_current();
-        {
-            let mut canvas = canvas.canvas.borrow_mut();
-
-            canvas.set_size(width as u32, height as u32, 1.0);
-
-            let image_id = canvas.create_image(image_ref, femtovg::ImageFlags::empty()).unwrap();
-            let mut path = femtovg::Path::new();
-            path.rect(0., 0., image_ref.width() as _, image_ref.height() as _);
-
-            let fill_paint = femtovg::Paint::image(
-                image_id,
-                0.,
-                0.,
-                image_ref.width() as _,
-                image_ref.height() as _,
-                0.0,
-                1.0,
-            );
-            canvas.fill_path(&mut path, fill_paint);
-            canvas.flush();
-            canvas.delete_image(image_id);
-        }
+        let mut softbuffer_buffer = Vec::with_capacity(width * height * 4);
+        softbuffer_buffer.extend(
+            buffer
+                .iter()
```

Comment:
if you do into_iter().map(...).collect()   Vec is able to keep the same buffer under some condition